quickwit/quickwit-rest-client/src/rest_client.rs

```diff
@@ -154,10 +156,24 @@ impl QuickwitClient {
         };
         while let Some(batch) = batch_reader.next_batch().await? {
             loop {
-                let response = self
-                    .transport
-                    .send::<()>(Method::POST, &ingest_path, None, None, Some(batch.clone()))
-                    .await?;
+                let response = if let (Some(query_param), false) = (
+                    last_block_commit.to_query_parameter(),
+                    batch_reader.has_next(),
+                ) {
+                    self.transport
+                        .send::<()>(
+                            Method::POST,
+                            &format!("{ingest_path}?{query_param}"),
```

Comment:
you should use the `query_string` argument that is used for passing your query parameters.

Reply:
I would simplify the code like this:

```rust

let query_params = if !batch_reader.has_next() {
    last_block_commit.to_query_parameter()
};
self.transport.send::<()>...
```